addon/list-utils.js

```diff
@@ -0,0 +1,69 @@
+import * as utils from './utils'
+import Ember from 'ember'
+
+function promiseValue (value) {
+  return new Ember.RSVP.Promise(function (resolve) {
+    resolve(value)
+  })
+}
+
+/**
+ * setOptions - set a list's available options
+ * @param  {Object} modelDef the bunsen model definition
+ * @param  {Object} dbStore the ember-data store
+ * @returns {Promise} a promise that resolves to a list of items
+ */
+export function getOptions (modelDef, dbStore) {
+  const enumDef = modelDef.enum
+  const queryDef = modelDef.query
+  let result
+  if (enumDef) {
+    result = promiseValue(getEnumValues(enumDef))
+  } else if (queryDef) {
+    result = getAsyncDataValues(modelDef, queryDef, dbStore)
+  } else {
+    result = promiseValue([])
+  }
+  return result
+}
+
+/**
+ * getEnumValues - take enum values and make into a collection
+ * @param  {String[]} values the values to make into a collection
+ * @returns {Object[]} a list of item objects
+ */
+export function getEnumValues (values = []) {
+  return values.map((value) => {
+    const label = Ember.String.capitalize(
+      value.split('-').join(' ').toLowerCase()
+    )
+    return {label, value}
+  })
+}
+
+/**
+ * setAsyncDataValues - Fetch the list of network functions from the backend and set them
+ * @param  {Object} modelDef the full bunsen model def for this property
+ * @param  {Object} queryDef the query obj from the model def
+ * @param  {Object} dbStore the ember-data store
+ * @returns {Promise} a promise that resolves to the list of items
+ */
+export function getAsyncDataValues (modelDef, queryDef, dbStore) {
+  const query = utils.createOrchQuery(queryDef)
```

Comment:
It looks like you're assuming that all queries will be in orch style. Is that a valid assumption? So no one will ever want to use this feature with a different API? It seems like making this *only* work for APIs that use a `q` and `p` query param with `:` separated queries is quite limiting. I don't think most APIs in the wild work that way. Most of the time it'll just be regular query params, right? `&foo=bar&baz=${foo.bar.baz}`? Shouldn't we just look for instances of `${}` and replace them with the proper values instead of needing to spit on `:`? Maybe I'm reading it wrong...